### DISCUSSION TITLE
error: Could not get shadow information for NOUSER

### DISCUSSION THREAD

**Discussion by piiskop** [rabbitmq/rabbitmq-server#2727] [Other] [answered]
I use ONLYOFFICE and ONLYOFFICE uses _rabbit-mq.server_. Every time I restart the server machine where both run, I have to purge and reinstall _rabbit-mq.server_ as it won't restart. Therefore, I first [reported the issue for ONLYOFFICE maintainers](https://github.com/ONLYOFFICE/DocumentServer/issues/1129). From there, I got the suggestion to report it to you.

**Comment by ShockwaveNN** [comment]
> I got the suggestion to report it to you

I suggested that because seems that stopping ONLYOFFICE DocumentServer alltogether and restarting machine without running DocumentServer will cause this error still, so I think it's something wrong on rabbitmq side

**Comment by michaelklishin** [comment]
See [RabbitMQ logs](https://www.rabbitmq.com/logging.html) and `journalctl --system` for clues as to what prevents the node from starting. We cannot suggest anything with the amount of information provided so far.

I don't see how

```
error: Could not get shadow information for NOUSER
```

is related. This is something logged by one of the OS services. You can see what RabbitMQ systemd service unit does as it is [open source](https://github.com/rabbitmq/rabbitmq-server-release/tree/master/packaging).

**Comment by piiskop** [comment, marked as answer]
> kalmer@test:/var/log/rabbitmq$ l -lat
> total 20
> -rw-r--r--  1 rabbitmq rabbitmq 3260 jaan  20 12:24 'rabbit@test.log'
> -rw-r--r--  1 rabbitmq rabbitmq  342 jaan  20 12:13  startup_log
> -rw-r--r--  1 rabbitmq rabbitmq   61 jaan  20 12:13  startup_err
> drwxr-xr-x  2 rabbitmq rabbitmq 4096 jaan  20 12:13  ./
> -rw-r--r--  1 rabbitmq rabbitmq    0 jaan  20 12:13 'rabbit@test-sasl.log'
> drwxrwxr-x 32 root     syslog   4096 jaan  20 12:13  ../
> kalmer@test:/var/log/rabbitmq$ tail rabbit@test.log 
> accepting AMQP connection <0.431.0> (127.0.0.1:60996 -> 127.0.0.1:5672)
> 
> =INFO REPORT==== 20-Jan-2021::12:24:58 ===
> connection <0.431.0> (127.0.0.1:60996 -> 127.0.0.1:5672): user 'guest' authenticated and granted access to vhost '/'
> 
> =INFO REPORT==== 20-Jan-2021::12:24:58 ===
> accepting AMQP connection <0.452.0> (127.0.0.1:60998 -> 127.0.0.1:5672)
> 
> =INFO REPORT==== 20-Jan-2021::12:24:58 ===
> connection <0.452.0> (127.0.0.1:60998 -> 127.0.0.1:5672): user 'guest' authenticated and granted access to vhost '/'
> kalmer@test:/var/log/rabbitmq$ tail startup_log 
> 
>               RabbitMQ 3.6.10. Copyright (C) 2007-2017 Pivotal Software, Inc.
>   ##  ##      Licensed under the MPL.  See http://www.rabbitmq.com/
>   ##  ##
>   ##########  Logs: /var/log/rabbitmq/rabbit@test.log
>   ######  ##        /var/log/rabbitmq/rabbit@test-sasl.log
>   ##########
>               Starting broker...
>  completed with 0 plugins.
> kalmer@test:/var/log/rabbitmq$ tail startup_err 
> systemd unit for activation check: "rabbitmq-server.service"

> kalmer@test:/var/log/rabbitmq$ sudo systemctl restart rabbitmq-server.service 
> kalmer@test:/var/log/rabbitmq$ sudo systemctl status rabbitmq-server.service 
> ● rabbitmq-server.service - RabbitMQ Messaging Server
>    Loaded: loaded (/lib/systemd/system/rabbitmq-server.service; enabled; vendor preset: enabled)
>    Active: active (running) since Wed 2021-01-20 13:02:04 EET; 1min 5s ago
>   Process: 19479 ExecStop=/usr/sbin/rabbitmqctl stop (code=exited, status=0/SUCCESS)
>  Main PID: 19667 (beam.smp)
>    Status: "Initialized"
>     Tasks: 163 (limit: 4915)
>    CGroup: /system.slice/rabbitmq-server.service
>            ├─19654 /bin/sh /usr/sbin/rabbitmq-server
>            ├─19667 /usr/lib/erlang/erts-9.2/bin/beam.smp -W w -A 128 -P 1048576 -t 5000000 -stbt db -zdbbl 32000 -K true -- -root /usr/lib/erl
>            ├─19766 /usr/lib/erlang/erts-9.2/bin/epmd -daemon
>            ├─19996 erl_child_setup 65536
>            ├─20030 inet_gethost 4
>            └─20031 inet_gethost 4
> 
> jaan  20 13:02:01 test.tennis24.ee systemd[1]: Starting RabbitMQ Messaging Server...
> jaan  20 13:02:04 test.tennis24.ee systemd[1]: rabbitmq-server.service: Supervising process 19667 which is not our child. We'll most likely no
> jaan  20 13:02:04 test.tennis24.ee systemd[1]: Started RabbitMQ Messaging Server.
> 

However after restarting the whole machine, _rabbitmq-server.service_ won't be able to start again itself and I have to reinstall it every single time.

So I restarted the machine for you again and used your suggested command by filtering in "abbit" because otherwise, it started to show a superlong log from the last year. I only copy the last lines for today here:

> jaan  20 13:13:02 test.tennis24.ee systemd[1]: rabbitmq-server.service: Scheduled restart job, restart counter is at 30.
> jaan  20 13:13:02 test.tennis24.ee systemd[1]: Stopped RabbitMQ Messaging Server.
> jaan  20 13:13:02 test.tennis24.ee systemd[1]: Starting RabbitMQ Messaging Server...
> jaan  20 13:13:02 test.tennis24.ee systemd[1]: rabbitmq-server.service: Main process exited, code=exited, status=1/FAILURE
> jaan  20 13:13:02 test.tennis24.ee systemd[1]: rabbitmq-server.service: Failed with result 'exit-code'.
> jaan  20 13:13:02 test.tennis24.ee systemd[1]: Failed to start RabbitMQ Messaging Server.
> jaan  20 13:13:12 test.tennis24.ee systemd[1]: rabbitmq-server.service: Service hold-off time over, scheduling restart.
> jaan  20 13:13:12 test.tennis24.ee systemd[1]: rabbitmq-server.service: Scheduled restart job, restart counter is at 31.
> jaan  20 13:13:12 test.tennis24.ee systemd[1]: Stopped RabbitMQ Messaging Server.
> jaan  20 13:13:12 test.tennis24.ee systemd[1]: Starting RabbitMQ Messaging Server...
> jaan  20 13:13:12 test.tennis24.ee systemd[1]: rabbitmq-server.service: Main process exited, code=exited, status=1/FAILURE
> jaan  20 13:13:12 test.tennis24.ee systemd[1]: rabbitmq-server.service: Failed with result 'exit-code'.
> jaan  20 13:13:12 test.tennis24.ee systemd[1]: Failed to start RabbitMQ Messaging Server.
> jaan  20 13:13:22 test.tennis24.ee systemd[1]: rabbitmq-server.service: Service hold-off time over, scheduling restart.
> jaan  20 13:13:22 test.tennis24.ee systemd[1]: rabbitmq-server.service: Scheduled restart job, restart counter is at 32.
> jaan  20 13:13:22 test.tennis24.ee systemd[1]: Stopped RabbitMQ Messaging Server.
> jaan  20 13:13:22 test.tennis24.ee systemd[1]: Starting RabbitMQ Messaging Server...
> jaan  20 13:13:22 test.tennis24.ee systemd[1]: rabbitmq-server.service: Main process exited, code=exited, status=1/FAILURE
> jaan  20 13:13:22 test.tennis24.ee systemd[1]: rabbitmq-server.service: Failed with result 'exit-code'.
> jaan  20 13:13:22 test.tennis24.ee systemd[1]: Failed to start RabbitMQ Messaging Server.

As you see, _RabbitMQ Messaging Server_ is unable to restart after the machine restart and keeps retrying. So I took a look at _RabbitMQ_'s logs again:

> kalmer@test:~$ cd /var/log/rabbitmq/
> kalmer@test:/var/log/rabbitmq$ l -lat
> total 24
> -rw-r--r--  1 rabbitmq rabbitmq    71 jaan  20 13:16  startup_log
> -rw-r--r--  1 rabbitmq rabbitmq     0 jaan  20 13:16  startup_err
> -rw-r--r--  1 rabbitmq rabbitmq 10444 jaan  20 13:04 'rabbit@test.log'
> drwxr-xr-x  2 rabbitmq rabbitmq  4096 jaan  20 12:13  ./
> -rw-r--r--  1 rabbitmq rabbitmq     0 jaan  20 12:13 'rabbit@test-sasl.log'
> drwxrwxr-x 32 root     syslog    4096 jaan  20 12:13  ../
> kalmer@test:/var/log/rabbitmq$ tail startup_log 
> ERROR: epmd error for host test: address (cannot connect to host/port)
> kalmer@test:/var/log/rabbitmq$ tail startup_err 
> kalmer@test:/var/log/rabbitmq$ tail rabbit@test.log 
>     asn1
>     xmerl
>     compiler
>     syntax_tools
>     sasl
>     stdlib
>     kernel
> 
> =INFO REPORT==== 20-Jan-2021::13:04:24 ===
> SIGTERM received - shutting down

Regardless of the status of the software, I'm not a developer of _RabbitMQ_. I just want to run ONLYOFFICE smoothlessly. It uses _RabbitMQ_ and _RabbitMQ_ fails to restart after the machine restart. So I don't understand why you closed the issue and locked my out for a while as I don't have skills to get over this problem. Please consider me as an end-user who has no administrative skills besides like _sudo apt install_ and _sudo apt purge_. You need to instruct me precisely.

> kalmer@test:/var/log/rabbitmq$ sudo rabbitmqctl status
> Status of node rabbit@test
> Error: unable to connect to node rabbit@test: nodedown
> 
> DIAGNOSTICS
> ===========
> 
> attempted to contact: [rabbit@test]
> 
> rabbit@test:
>   * unable to connect to epmd (port 4369) on test: address (cannot connect to host/port)
> 
> 
> current node details:
> - node name: 'rabbitmq-cli-93@test'
> - home dir: /var/lib/rabbitmq
> - cookie hash: LdDZdDghXOODLiAbu6vf5w==

> kalmer@test:/var/log/rabbitmq$ sudo netstat -plunt|grep :4369
> tcp        0      0 0.0.0.0:4369            0.0.0.0:*               LISTEN      22584/epmd          
> tcp6       0      0 :::4369                 :::*                    LISTEN      22584/epmd          
> kalmer@test:/var/log/rabbitmq$ sudo ps -eF|grep 22584
> rabbitmq 22584     1  0  6716   204   7 13:23 ?        00:00:00 /usr/lib/erlang/erts-9.2/bin/epmd -daemon
> kalmer   25132  3076  0  4335  1148   6 13:26 pts/0    00:00:00 grep --color=auto 22584

Then, I started to mess around in _rabbitmq_'s configuration file with some other results but with no success. Finally, I followed a suggestion at Programmersought:

> kalmer@test:/var/lib/rabbitmq$ sudo emacs /etc/hosts
> 27.0.0.1       localhost test
> 

> kalmer@test:/var/lib/rabbitmq$ sudo systemctl restart rabbitmq-server.service 
> almer@test:/var/lib/rabbitmq$ sudo rabbitmqctl status
> Status of node rabbit@test
> [{pid,3460},
>  {running_applications,
>      [{rabbit,"RabbitMQ","3.6.10"},
>       {ranch,"Socket acceptor pool for TCP protocols.","1.3.0"},
>       {ssl,"Erlang/OTP SSL application","8.2.3"},
>       {public_key,"Public key infrastructure","1.5.2"},
>       {crypto,"CRYPTO","4.2"},
>       {asn1,"The Erlang ASN1 compiler version 5.0.4","5.0.4"},
>       {rabbit_common,
>           "Modules shared by rabbitmq-server and rabbitmq-erlang-client",
>           "3.6.10"},
>       {xmerl,"XML parser","1.3.16"},
>       {compiler,"ERTS  CXC 138 10","7.1.4"},
>       {os_mon,"CPO  CXC 138 46","2.4.4"},
>       {mnesia,"MNESIA  CXC 138 12","4.15.3"},
>       {syntax_tools,"Syntax tools","2.1.4"},
>       {sasl,"SASL  CXC 138 11","3.1.1"},
>       {stdlib,"ERTS  CXC 138 10","3.4.3"},
>       {kernel,"ERTS  CXC 138 10","5.4.1"}]},
>  {os,{unix,linux}},
>  {erlang_version,
>      "Erlang/OTP 20 [erts-9.2] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:128] [kernel-poll:true]\n"},
>  {memory,
>      [{total,56831616},
>       {connection_readers,0},
>       {connection_writers,0},
>       {connection_channels,0},
>       {connection_other,0},
>       {queue_procs,2744},
>       {queue_slave_procs,0},
>       {plugins,0},
>       {other_proc,21780240},
>       {mnesia,61680},
>       {metrics,184664},
>       {mgmt_db,0},
>       {msg_index,40872},
>       {other_ets,1757112},
>       {binary,37568},
>       {code,21269571},
>       {atom,891849},
>       {other_system,10987172}]},
>  {alarms,[]},
>  {listeners,[{clustering,25672,"::"},{amqp,5672,"::"}]},
>  {vm_memory_high_watermark,0.4},
>  {vm_memory_limit,5030816972},
>  {disk_free_limit,50000000},
>  {disk_free,72108466176},
>  {file_descriptors,
>      [{total_limit,65436},
>       {total_used,2},
>       {sockets_limit,58890},
>       {sockets_used,0}]},
>  {processes,[{limit,1048576},{used,160}]},
>  {run_queue,0},
>  {uptime,9},
>  {kernel,{net_ticktime,60}}]
 
So it works for now without having to reinstall _rabbitmq-server_. And it also works after machine restart. You need to change something in your software so that its users know how to get it work. The installer should check for the hostname.

**Reply by michaelklishin** [reply]
I moved your issue to a discussion, a relatively new GitHub feature that is separate from issues. Our team does not use GitHub issues for questions and investigations. When I convert an issue to a discussion, GitHub closes it first. I understand that this is confusing but we have no control over it.

I did not realize that GitHub now also **locks** issues when they are converted. This will make maintainers who use this feature look really bad even though all they did is hit the "Convert to discussion" button. Submitted some feedback to GitHub about this https://github.com/github/feedback/discussions/3197.

**Comment by michaelklishin** [comment]
@piiskop I moved your issue to a discussion, a relatively new GitHub feature that is separate from issues. Our team does not use GitHub issues for questions and investigations. When I convert an issue to a discussion, GitHub closes it first. I understand that this is confusing but we have no control over it.

Indeed RabbitMQ (or rather its runtime) needs to be able to resolve [its own hostname](https://www.rabbitmq.com/clustering.html#cluster-formation-requirements), even if it's `localhost`. So your solution makes sense.

It is worth mentioning that ONLYOFFICE runs a version of RabbitMQ that has gone entirely out of support [in May 2018](https://www.rabbitmq.com/versions.html). They may choose to upgrade.

**Reply by piiskop** [reply]
@ShockwaveNN please see the allegation of @michaelklishin on outdated _RabbitMQ_ for _ONLYOFFICE_. I told you it was unreliable to use third-party software.

**Reply by michaelklishin** [reply]
There are no allegations. The version is mentioned in the log and `rabbitmqctl status` output. Let's not get too far with conclusions about "3rd party software" as virtually all software has dependencies of some kind. Dependencies have their own maintenance schedule and have to be [upgraded](https://www.rabbitmq.com/upgrade.html) from time to time.

**Reply by ShockwaveNN** [reply]
@piiskop Since you're using *.deb version of onlyoffice - you're using rabbitmq provided by your system, I can presume your using this one
https://packages.ubuntu.com/bionic-updates/rabbitmq-server, which one is 3.6.10 on Ubuntu 18.04

**Reply by michaelklishin** [reply]
That might very well be the case. Here's how to [install a modern RabbitMQ version on Debian and Ubuntu](https://www.rabbitmq.com/install-debian.html). Good catch @ShockwaveNN.

**Reply by piiskop** [reply]
Yes, this thought just came to my mind, too. I'm currently upgrading my server to _20.04_.

**Comment by michaelklishin** [comment]
We did not lock you out or limit your ability to respond. When an issue is converted to a discussion, its comments won't be transferred for a period of time. I suspect that the ability to post new comments may be temporarily unavailable. My apologies about that confusion but this is how GitHub discussions work right now.

**Comment by michaelklishin** [comment]
Wow, I did not realize that GitHub now closes *and locks* issues when they are converted. This will make maintainers who use this feature look really bad even though all they did is hit the "Convert to discussion" button.
